Add `eDisMax` compatibility for filter queries

alexandrevryghem opened the pull request:
## Description
In DSpace, we use the `DisMax` query parser for Solr. However, when you switch to [`eDisMax`](https://solr.apache.org/guide/6_6/the-extended-dismax-query-parser.html), some functionality and tests fail. This is because `eDisMax` doesn't allow syntax like `subject: subject1`; it only supports `subject:subject1`. Since `DisMax` supports both, it might be worth adding these fixes to prevent others from encountering the same issue. 😅

## Instructions for Reviewers
List of changes in this PR:
* Removed spaces around `:` in filter queries to prevent the `eDisMax` query parser from breaking
* Also fixed a `DiscoveryIT` test that can fail due to its reliance on the order of returned items

**Guidance for how to test or review this PR:**
Verify that everything still works like expected

## Checklist
- [x] My PR is **created against the `main` branch** of code (unless it is a backport or is fixing an issue specific to an older branch).
- [x] My PR is **small in size** (e.g. less than 1,000 lines of code, not including comments & integration tests). Exceptions may be made if previously agreed upon.
- [x] My PR **passes Checkstyle** validation based on the [Code Style Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Style+Guide).
- [x] My PR **includes Javadoc** for _all new (or modified) public methods and classes_. It also includes Javadoc for large or complex private methods.
- [x] My PR **passes all tests and includes new/updated Unit or Integration Tests** based on the [Code Testing Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Testing+Guide).
- [x] My PR **includes details on how to test it**. I've provided clear instructions to reviewers on how to successfully test this fix or feature.
- [x] If my PR includes new libraries/dependencies (in any `pom.xml`), I've made sure their licenses align with the [DSpace BSD License](https://github.com/DSpace/DSpace/blob/main/LICENSE) based on the [Licensing of Contributions](https://wiki.lyrasis.org/display/DSPACE/Code+Contribution+Guidelines#CodeContributionGuidelines-LicensingofContributions) documentation.
- [x] If my PR modifies REST API endpoints, I've opened a separate [REST Contract](https://github.com/DSpace/RestContract/blob/main/README.md) PR related to this change.
- [x] If my PR includes new configurations, I've provided basic technical documentation in the PR itself.
- [x] If my PR fixes an issue ticket, I've [linked them together](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue).

alexandrevryghem committed:
111162: Removed spaces before solr query values (because some parsers don't support this)

nwoodward commented:
Hi @alexandrevryghem. These changes look good, and it would be nice to switch to the eDisMax query parser. Is there an easy way to test compatibility of the queries, such as adding "defType=edismax" to every query? How should I test it?

saschaszott commented:
@alexandrevryghem, the Solr core `search` in DSpace (8.1) is still using the Standard Query Parser.

I noticed this in the Solr configuration:

https://github.com/DSpace/DSpace/blob/474becc847b01f7245781108306acd2ee616e5ef/dspace/solr/search/conf/solrconfig.xml#L116

Have you switched to the DisMax or eDisMax query parser locally?

I’ve created GitHub issue #10494 today to address this transition in the DSpace core.

saschaszott commented:
@alexandrevryghem : I've successfully integrated your changes in my local DSpace instance (where I currently use edismax). How are the changes in `DiscoveryIT` related to this PR?

alexandrevryghem commented:
@saschaszott: It was the `eDisMax` query parser, but I wasn’t the one who set it up. This PR only addresses the issues it caused with DSpace. I thought it could be useful for others who wanted to switch to eDisMax as well, but it’s been over a year since I last worked with that code, so I don’t remember much. Regarding the changes to `DiscoveryIT`, they fixed an issue where the sort order of the results was altered due to configuration changes in `solrconfig.xml`. If you prefer, I can leave out those changes.

saschaszott commented:
@alexandrevryghem , thanks for your feedback! I’m happy to approve this PR if you remove the changes in DiscoveryIT.

alexandrevryghem committed:
Merge branch 'add-edismax-parser-support_contribute-7.6'

# Conflicts:
#	dspace-api/src/main/java/org/dspace/statistics/SolrLoggerServiceImpl.java

saschaszott commented:
@tdonohue , do we need another reviewer for this PR, or can it still be included in DS9? This PR does **not** enable edismax, it contains only syntactic changes (removal of obsolete whitespaces in filter queries).

tdonohue reviewed (approved):
👍 Thanks @alexandrevryghem !  Looks good to me too.  Thanks also to @saschaszott for noting this is just whitespace changes.
